File: app/src/main/java/com/thewizrd/simpleweather/preferences/WindowColorPreferenceFragmentCompat.java
package com.thewizrd.simpleweather.preferences;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.LifecycleOwner;

import com.thewizrd.shared_resources.helpers.OnBackPressedFragmentListener;
import com.thewizrd.simpleweather.helpers.WindowColorManager;

public abstract class WindowColorPreferenceFragmentCompat extends CustomPreferenceFragmentCompat
        implements OnBackPressedFragmentListener, WindowColorManager {

    @Override
    public boolean onBackPressed() {
        return false;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getLifecycle().addObserver(new DefaultLifecycleObserver() {
            @Override
            public void onStart(@NonNull LifecycleOwner owner) {
                DefaultLifecycleObserver.super.onStart(owner);
                updateWindowColors();
            }
        });
    }

    public void updateWindowColors() {
    }
}

File: app/src/main/java/com/thewizrd/simpleweather/preferences/WindowColorPreferenceFragmentCompat.kt
package com.thewizrd.simpleweather.preferences

import android.os.Bundle
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.thewizrd.simpleweather.helpers.WindowColorManager
import kotlinx.coroutines.launch

abstract class WindowColorPreferenceFragmentCompat : CustomPreferenceFragmentCompat(),
    WindowColorManager {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                updateWindowColors()
            }
        }
    }

    override fun updateWindowColors() {}
}